### PR TITLE
Add GH workdlow actions for Unit and Sanity tests

### PR DESCRIPTION
**Pull request by justjais**
Signed-off-by: Sumit Jaiswal <sjaiswal@redhat.com>

##### SUMMARY

Add GH workdlow actions for Unit and Sanity tests

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

asa

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by justjais** [comment]
recheck

**Comment by justjais** [comment]
recheck

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -11,11 +11,11 @@ repos:
         args: [--branch, main]
       - id: trailing-whitespace
   - repo: https://github.com/psf/black
-    rev: 19.3b0
+    rev: 22.3.0
     hooks:
       - id: black
         args: [-l, "79"]
```
Comment (KB-perByte) [inline .pre-commit-config.yaml:17]:
```suggestion
        args: [-l, "100"]
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[tool.black]
+line-length = 79
```
Comment (KB-perByte) [inline pyproject.toml:2]:
```suggestion
line-length = 100
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
+
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  changelog:
```
Comment (KB-perByte) [inline .github/workflows/test.yml:14]: we might consider adding ansible-lint

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.1.0
+    rev: v4.2.0
```
Comment (KB-perByte) [inline .pre-commit-config.yaml:4]:
```suggestion
    rev: v4.3.0
```

**Review by KB-perByte — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -11,11 +11,11 @@ repos:
         args: [--branch, main]
       - id: trailing-whitespace
   - repo: https://github.com/psf/black
-    rev: 19.3b0
+    rev: 22.3.0
```
Comment (KB-perByte) [inline .pre-commit-config.yaml:14]:
```suggestion
    rev: 22.6.0
```

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ---
```
Comment (KB-perByte) [inline .pre-commit-config.yaml:1]: we might also add/enable isort and add-tailing-comma to the pre-commit CI
ref : https://github.com/ansible-collections/cisco.ios/blob/main/.pre-commit-config.yaml

**Review by KB-perByte — APPROVED** [review]

**Pull request merged**